backend/tests/db_integration/test_db_integration.py

```diff
@@ -0,0 +1,36 @@
+from testcontainers.mongodb import MongoDbContainer
+from pymongo import MongoClient
+import pytest
+
+
+@pytest.fixture(scope="session")
+def mongo_client() -> MongoClient:
+    with MongoDbContainer() as mongo:
+        yield mongo.get_connection_client()
+
+
+def test_mongo_insert_and_find(mongo_client: MongoClient):
+    db = mongo_client.first_instance
```

Comment:
If you're manually creating/choosing instances in each test, it still allows for unwanted interferences. `mongo_client` should return a unique instance, preferably named after the test it's being used in. This could be implemented with the addition of a second fixture, that uses previously created MongoDbContainer.
_This may be helpful: https://docs.pytest.org/en/4.6.x/reference.html#request_